src/user-guide/writing-workflows/runtime.rst

```diff
@@ -317,6 +319,25 @@ output files are typically held in cycle point sub-directories of this.
 The top level share directory location can be changed, e.g. to a large data
 area, by global config settings under :cylc:conf:`global.cylc[install][symlink dirs]`.
 
+If your workflow creates or installs executables or Python libraries
+as it is running, these can be placed in:
+
+* ``share/bin`` - for executables.
+* ``share/lib/python`` - for Python modules.
+
+These folders will be added to ``PATH`` and ``PYTHONPATH`` so they can be
+used by jobs in the same way as
+:ref:`top level "bin/" and "lib/python/" directories <WorkflowDefinitionDirectories>`.
+
+.. note::
+
+   Cylc will not create these folders.
+
+.. important::
+
+  These locations will be placed first in the executable and Python module
+  search paths, before those in the parent run directory .
```

Comment:
Maybe
```suggestion
   These locations take precedence over the top level ``bin/`` and
   ``lib/python/`` directories in the run dir.
```

Reply:
@wxtim and @MetRonnie - this kind of reverse my 2nd review suggestion above.  I would prefer to just "state the facts" here, which is where these directories are put in the search paths.  "Take precedence" or "called first" (Tim's original phrasing) could be assumed to have wider implications, e.g. for how Cylc itself somehow manages these locations.

Reply:
I rather agree with Hilary - can we come to a conclusion @MetRonnie ?

Reply:
https://github.com/cylc/cylc-doc/pull/609/files#r1211855271